hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -2303,12 +2363,14 @@ public void markRegionAsSplit(final RegionInfo parent, final ServerName serverNa
   public void markRegionAsMerged(final RegionInfo child, final ServerName serverName,
     RegionInfo[] mergeParents) throws IOException {
     final RegionStateNode node = regionStates.getOrCreateRegionStateNode(child);
-    node.setState(State.MERGED);
     for (RegionInfo ri : mergeParents) {
       regionStates.deleteRegion(ri);
+      regionInTransitionTracker.handleRegionDelete(ri);
     }
+    // TODO need to handle delete and new region
```

Comment:
Is this comment still relevant? Looks like the delete and new region creations are handled.